tensorflow/contrib/learn/python/learn/datasets/mnist.py

```diff
@@ -223,23 +223,25 @@ def fake():
   TEST_IMAGES = 't10k-images-idx3-ubyte.gz'
   TEST_LABELS = 't10k-labels-idx1-ubyte.gz'
 
-  local_file = base.maybe_download(TRAIN_IMAGES, train_dir,
-                                   SOURCE_URL + TRAIN_IMAGES)
+  def download_local_file(fname):
+    try:
+      return base.maybe_download(fname, train_dir, SOURCE_URL + fname)
```

Comment:
Hmm, `maybe_download` is defined in a different file that I think is used by many other datasets, and its current method signature doesn't really allow for the concept of a backup URL. Would you suggest an optional `backup` keyword argument?